From: Ondrej Jirman <megous@megous.com>
Date: Mon, 15 Nov 2021 04:09:50 +0100
Subject: [PATCH 01/36] base: property: Swap order of search for connection to
 "devcon, graph"

This avoids confusing error in kernel log, when using devcon matching
in DT. Example:

"OF: graph: no port node found in /i2c@ff3d0000/typec-portc@22"

This suggest there's some error because graph based search failed,
but there is no error, because devcon based matching succeeds.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/base/property.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/base/property.c b/drivers/base/property.c
index ed6f449..cbed556 100644
--- a/drivers/base/property.c
+++ b/drivers/base/property.c
@@ -1292,11 +1292,11 @@ void *fwnode_connection_find_match(struct fwnode_handle *fwnode,
 	if (!fwnode || !match)
 		return NULL;
 
-	count = fwnode_graph_devcon_matches(fwnode, con_id, data, match, &ret, 1);
+	count = fwnode_devcon_matches(fwnode, con_id, data, match, &ret, 1);
 	if (count)
 		return ret;
 
-	count = fwnode_devcon_matches(fwnode, con_id, data, match, &ret, 1);
+	count = fwnode_graph_devcon_matches(fwnode, con_id, data, match, &ret, 1);
 	return count ? ret : NULL;
 }
 EXPORT_SYMBOL_GPL(fwnode_connection_find_match);
